core/core/src/main/java/androidx/core/app/ActivityCompat.java

```diff
@@ -553,7 +555,9 @@ public void run() {
      */
     public static boolean shouldShowRequestPermissionRationale(@NonNull Activity activity,
             @NonNull String permission) {
-        if (Build.VERSION.SDK_INT >= 23) {
+        if (Build.VERSION.SDK_INT == 31) {
```

Comment:
> I assume we dont want this workaround in perpetuity (SDK_INT > 31). Should this first check for SDK_INT >= 32 can call in without a workaround?

SDK_INT >= 32 does not have this problem, because this problem has been fixed in Android 12L

Reply:
Does this mean this code should look like 

```
if Build.VERSION.SDK_INT >= 32 {
             return Api32Impl.shouldShowRequestPermissionRationale(activity, permission);
} else if if (Build.VERSION.SDK_INT == 31) {
            return Api31Impl.shouldShowRequestPermissionRationale(activity, permission);
} else if (Build.VERSION.SDK_INT >= 23) {
            return Api23Impl.shouldShowRequestPermissionRationale(activity, permission);
}
```
?

Reply:
@liutikas Hello, the code has been modified as required.